src/support/assistant-support.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { FirefallClient } from '@adobe/spacecat-shared-gpt-client';
+import { ErrorWithStatusCode } from './utils.js';
+
+/*
+ * Command Config Type:
+ *  - parameters: string[] : parameters required for the command
+ *  - responseFormat: string : AI response format for the command
+ *  - llmModel: string : the LLM Model to use for the command
+ */
+const commandConfig = {
+  findMainContent: {
+    parameters: [],
+    llmModel: 'gpt-4-turbo',
+    responseFormat: 'json_object',
+  },
+  findRemovalSelectors: {
+    parameters: ['prompt'],
+    llmModel: 'gpt-4-turbo',
+    responseFormat: 'json_object',
+  },
+  findBlockSelectors: {
+    parameters: ['prompt', 'imageUrl'],
+    llmModel: 'gpt-4-vision',
+  },
+  findBlockCells: {
+    parameters: ['prompt', 'imageUrl'],
+    llmModel: 'gpt-4-vision',
+  },
+  generatePageTransformation: {
+    parameters: ['prompt'],
+    llmModel: 'gpt-4-turbo',
+    responseFormat: 'json_object',
+  },
+};
+
+const SCOPE = {
+  ASSISTANT: 'imports.assistant', // allows users submit prompts with their API key
+};
+
+const STATUS = {
+  BAD_REQUEST: 400,
+  UNAUTHORIZED: 401,
+  OK: 200,
+};
+
+/**
+ * Verify that the authenticated user has the required level of access scope.
+ * @param auth the auth object from the context.
+ * @param scopes a list of scopes to validate the user has access to.
+ * @param log optional logger.
+ * @return {true} if scope is allowed.
+ */
+const validateAccessScopes = (auth, scopes, log) => {
+  log?.debug(`validating scopes: ${scopes}`);
+
+  try {
+    auth.checkScopes(scopes);
+  } catch (error) {
+    throw new ErrorWithStatusCode('Missing required scopes.', STATUS.UNAUTHORIZED);
+  }
+};
+
+const getFirefallCompletion = async (requestData, log) => {
```

Comment:
Since this is an async function should it be named `fetchFirefallCompletion`? Or is there an existing pattern you are following that uses the `get` prefix?

```suggestion
const fetchFirefallCompletion = async (requestData, log) => {
```